README.md

```diff
@@ -124,6 +125,33 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
+
+| client-go    | [apimachinery](https://github.com/kubernetes/apimachinery) | [apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver)<sup>*</sup> | [code-generator](https://github.com/kubernetes/code-generator)<sup>+</sup> |
+| ----------------------------------| -----------------------------------------------------------|----------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+| tag: v5.0.x  | tag: kubernetes-1.8.x  | tag: kubernetes-1.8.x | tag: kubernetes-1.8.x |
+| tag: v6.0.x  | tag: kubernetes-1.9.x  | tag: kubernetes-1.9.x | tag: kubernetes-1.9.x |
+
+<sup>*</sup> optional library that exposes the Extensions API, for example, for 
+Custom Resource Definitions.  
+<sup>+</sup> optional library for generating Kubernetes style API types. Typically 
+used with Custom Resource Definitions.
+
+Note that `x` versions need to align. For example tag v.5.0.1 in client-go, will be
+supported with tag kubernetes-1.8.1 in apimachinery and other repositories
+
+Using the git tag versions is the preferred way of managing dependencies for newer 
+releases of client-go, and since client-go is backward compatible, it is recommended that
+developers always use the latest version.
+
+If you are using [Glide](https://glide.sh) or [godep](https://github.com/tools/godep) 
+to manage your Go dependencies, this will manage the non-optional libraries automatically.
```

Comment:
godep is not automatic. Instead, you have to do `godep restore` from client-go and `godep save` from your 3rdparty code base.

I would prefer something along these lines:

Client-go specifies non-optional libraries in its [Godeps/Godeps.json](Godeps.json) file. While using glide will respect these automatically, with glide you have to manually `godep restore` them from client-go's directory and then `godep save` them in your project.

Reply:
And please add `dep`:

Note that golang/dep currently **does not respect the specified dependencies in client-go**. You have to manually add them as constraints in your `Gopkg.toml` for all its dependencies (not only for `k8s.io/api` and `k8s.io/apimachinery`!).

Reply:
Was that original comment meant to say:

> Client-go specifies non-optional libraries in its Godeps/Godeps.json file. While using glide will respect these automatically, with **godep** you have to manually godep restore them from client-go's directory and then godep save them in your project.

Is that correct?

Reply:
The one quoted? Yes.

Reply:
Updated and rebased. PTAL.